### PR TITLE
【Hackathon No.8】

### PR DESCRIPTION
**Pull request by Li-fAngyU**

### CODE REVIEW

File: rfcs/APIs/20220310_api_design_for_nanmean.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+```
+Computes the mean of all non-NaN elements along the specified dimensions.
+
+This function is identical to torch.mean() when there are no NaN values in the input tensor. In the presence of NaN, torch.mean() will propagate the NaN to the output whereas torch.nanmean() will ignore the NaN values (torch.nanmean(a) is equivalent to torch.mean(a[~a.isnan()])).
+
+If keepdim is True, the output tensor is of the same size as input except in the dimension(s) dim where it is of size 1. Otherwise, dim is squeezed (see torch.squeeze()), resulting in the output tensor having 1 (or len(dim)) fewer dimension(s).
+```
+
+### 实现方法
+由于没有找到Pytorch实现nanmeanAPI的源码，所以未能提供其核心代码，但通过其在document上的介绍[torch.nanmean](https://pytorch.org/docs/stable/generated/torch.nanmean.html#torch.nanmean),可以确定其整体逻辑与Numpy基本一致。
```
Comment (jeff41404) [inline rfcs/APIs/20220310_api_design_for_nanmean.md:82]: pytorch 相关源码在 https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/ReduceOps.cpp，可以补充下这部分的调研

Comment (Li-fAngyU) [reply rfcs/APIs/20220310_api_design_for_nanmean.md:82]: 好的谢谢！

**Comment by Li-fAngyU** [comment]
@Ligoml 你好！我已按照reviewer的建议完成nanmean_rcf文件在pytorch源码上的补充。感谢reviewer提供的建议！

File: rfcs/APIs/20220310_api_design_for_nanmean.md (hunk 1 above)
Comment (Li-fAngyU) [reply rfcs/APIs/20220310_api_design_for_nanmean.md]: @jeff41404 感谢您提供的建议以及pytorch资料，我已按照您的建议进行了修改。

File: rfcs/APIs/20220311_api_design_for_nanmean.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+测试考虑的case如下：
+
+- 和numpy结果的数值的一致性, `paddle.nanmean`,和`np.nanmean`结果是否一致；
+- 参数`axis`为int,tuple和list时输出的正确性；
+- `keepdim`参数的正确性；
+- 未输入维度时的输出正确性；
+- 输入含`NaN`结果的正确性；
+- 输入在指定轴上存在元素都为NaN时,结果的正确性；
+- 错误检查：输入`x`不是Tensor时,能否正确抛出错误；
+- 错误检查：`axis`所指维度在当前Tensor中不合法时能正确抛出错误。
```
Comment (jeff41404) [inline rfcs/APIs/20220311_api_design_for_nanmean.md:137]: 测试case中需要考虑测试反向梯度计算结果，包括nan值和非nan值位置的梯度
设计中其它部分已经考虑比较完善，加油！

Comment (Li-fAngyU) [reply rfcs/APIs/20220311_api_design_for_nanmean.md:137]: 感谢提醒！反向梯度计算结果的测试很重要，是我疏忽了。我去查阅一下pytoch的资料，然后在测试case上补充上相关的内容。

Comment (Li-fAngyU) [reply rfcs/APIs/20220311_api_design_for_nanmean.md:138]: 我已经按照您的建议对rcf文件里的测试case进行修改并补充，再次感谢您提供的建议，这对我有很大的帮助。
我有一个小困惑，即在反向梯度计算时如果对应位置为nan值，那么在反向梯度传播时是否是以nan值进行传播？望解惑。

Comment (jeff41404) [reply rfcs/APIs/20220311_api_design_for_nanmean.md:138]: 从计算逻辑上看，nan值对于结果没有贡献，所以理论上应该是 非nan值位置上有正常的梯度数值，nan值对应的梯度为0。可以做个实验也能看出来

Comment (Li-fAngyU) [reply rfcs/APIs/20220311_api_design_for_nanmean.md:138]: 我明白了。感谢！

Comment (Li-fAngyU) [reply rfcs/APIs/20220311_api_design_for_nanmean.md:138]: @jeff41404 我已对rcf文件里的测试case进行补充和修改，请检查是否符合要求。谢谢！

**Review by jeff41404 — APPROVED** [review]
LGTM
